### PR TITLE
MVP

### PR DESCRIPTION
**Pull request by adridev**
He here is my solution is a very basic but I don't know how It take me a lot of time. I can improve the inexistence business layer adding a command bus a a few commands to decouple Doctrine mapping from forms. I should add a few Unit test also. I will continue improving it. It you want please tell me and I do another pull request.

Thanks.
Adrián Ríos.

### CODE REVIEW

File: src/Kodify/BlogBundle/Resources/views/Post/view.html.twig [hunk 1]
```diff
@@ -9,5 +9,34 @@
...
+                    <div>
+                        <span class="comment-text">{{ comment.text }}</span>
+                    </div>
+                    <div class="comment-date">
+                        {{ comment.createdAt|date('d/m/Y H:i') }}
+                    </div>
+                </li>
+            {% else %}
+                <li class="no-comments">There are no comments</li>
+            {% endfor %}
```
Comment (marcostvz) [inline src/Kodify/BlogBundle/Resources/views/Post/view.html.twig:36]: Right now here you show up to 10 comments. How would you do an option to see more of them and what would you take into account? (There's no need to do it programmatically, just a quick explanation :smile:)

File: src/Kodify/BlogBundle/Resources/views/Post/view.html.twig [hunk 2]
```diff
@@ -9,5 +9,34 @@
...
+        </ul>
+        <footer>
+            <a href="{{ path('comment_post', {'id': post.id}) }}" class="new-comment btn btn-success">Comment</a>
```
Comment (marcostvz) [inline src/Kodify/BlogBundle/Resources/views/Post/view.html.twig:39]: If we wanted to put the comment box here, how would you do it?

File: src/Kodify/BlogBundle/Controller/CommentController.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            $em = $this->get('doctrine')->getManager();
+            $em->persist($comment);
+            $em->flush();
+
+            $postId = $comment->getPost()->getId();
+
+            return $this->redirectToRoute('view_post', ['id' => $postId]);
+        }
```
Comment (marcostvz) [inline src/Kodify/BlogBundle/Controller/CommentController.php:42]: Hey Adrián,

what if the form is not valid? What would you show to the user and how?

Cheers!

**Comment by marcostvz** [comment]
Overall you did a very nice PR :) We're willing to see you soon ^^

About the comments I did, if you can just answer them is ok, no need to program anything more!

File: src/Kodify/BlogBundle/Resources/views/Post/view.html.twig (hunk 1 above)
Comment (adridev) [inline src/Kodify/BlogBundle/Resources/views/Post/view.html.twig:36]: Hi,
We have at least 2 options, we can make an AJAX request to a controller that returns a full html view or a JSON(like an API), an API, or refresh the full page passing the filters, order, limit... in the query. Of course we can't retrieve post directly by lazy loading from post object. In one of the last  commits this is refactored to dont do it, and comments are passed from controlled already "filtered".

File: src/Kodify/BlogBundle/Resources/views/Post/view.html.twig (hunk 2 above)
Comment (adridev) [inline src/Kodify/BlogBundle/Resources/views/Post/view.html.twig:39]: The comment box should be there, I begin to care about time and didn't do it. Also, the sentence in the scenario "I press comment" and after, "I press publish" (or something like this) made my think that you want a two steps feature. But of course, you shouldn't leave the page to comment a post... This should be an AJAX feature. Also, we can do it in the same page without AJAX but I think AJAX is perfect for this use case.

File: src/Kodify/BlogBundle/Controller/CommentController.php (hunk 3 above)
Comment (adridev) [inline src/Kodify/BlogBundle/Controller/CommentController.php:42]: Sorry. I saw the variable message that you use in other cases, I thought in to make a refactor to use a generic notification system with session flashbag instead, but I should forgot it.

Comment (adridev) [inline src/Kodify/BlogBundle/Controller/CommentController.php:42]: Do you mean, when it is valid? In this case the user should see his message in the target of the redirect, is true that we can show a flash message also to improve the feedback. When is invalid the user should see the errors in the form page.

**Pull request closed without merging**